### ISSUE TITLE
Documentation indicates remove() method will run removed(node) callback

### ISSUE THREAD

**Issue by Flaniga3** [angular-ui-tree/angular-ui-tree#586]
Just wanted to let you know that the documentation on the main-page seems to indicate that the remove() method will run the removed(node) callback. When I noticed this wasn't the case I took a quick glance at the source code and noticed the method which runs the callback is actually removeNode(). Not really a huge issue, just seems like a simple documentation change.

Great plugin by the way.

**Comment by FrancoisMartin** [comment]
I don't have the same issue.
For me, ```removed``` is a valid callback.

```javascript
$scope.options = {
  removed: function(node) {
  	console.log(node);
  }
}
```

**Comment by Flaniga3** [comment]
Strange. Oh well, removeNode() works fine for me and if there are no known issues with using that instead I'm fine keeping it that way.

**Comment by rpalacios** [comment]
@Flaniga3's comment help me a lot.

I did use the [Basic Example](http://angular-ui-tree.github.io/angular-ui-tree/#/basic-example) to startup the integration to my web app. There you use `remove(this)` on delete button. Using it this way, it does not call the `removed(node)` callback.

Doing the change suggested by @Flaniga3 did the trick. So the remove button element should be:

    <a class="pull-right btn btn-danger btn-xs" data-nodrag ng-click="removeNode(this)" title="Remove">
        <span class="glyphicon glyphicon-remove"></span>
    </a>

This way, it works perfectly.

**Comment by Flaniga3** [comment]
@rpalacios really happy I could help out!

**Comment by Voles** [comment]
@rpalacios / @Flaniga3 great to see the issue is solved! Any chance any of you guys has the time to update the documentation / example in a PR?

**Comment by Flaniga3** [comment]
I actually closed by mistake and didn't realize until now.

**Comment by Flaniga3** [comment]
I still have no luck using the basic remove. I'd look into it more to try to help out, but I'm on a rather tight development schedule here.

**Comment by rpalacios** [comment]
I think I could do it this weekend.

**Comment by Flaniga3** [comment]
Is a simple documentation change really the right fix? I feel an effort to convince the "remove(node)" to initiate the call back would be a wise move.

**Comment by Aaronm14** [comment]
Agreed, this was pretty confusing, the only way to figure this out is digging into the source code.  I'm using removeNode myself and immediately searched through the issues to see if anyone else was having the same problem.  Is it a bug or a simple documentation update?  I'm not quite sure why remove wouldn't also fire off the callback or why they are separate functions to begin with.

**Comment by Voles** [comment]
This issue has been fixed by @IdanCo in #673. The `removeNode` method has been removed to prevent confusion.The fix will be included in the next release!
